plugins/by-name/quarto/default.nix

```
@@ -0,0 +1,111 @@
+{ lib, pkgs, ... }:
+let
+  inherit (lib.nixvim) defaultNullOpts;
+  inherit (lib) types;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "quarto";
+  package = "quarto-nvim";
+
+  maintainers = [ lib.maintainers.BoneyPatel ];
+
+  settingsOptions = {
+    debug = defaultNullOpts.mkBool false ''
+      Enables or disables debugging.
+    '';
+
+    closePreviewOnExit = defaultNullOpts.mkBool true ''
+      Closes preview on exit.
+    '';
+
+    lspFeatures.enabled = defaultNullOpts.mkBool true ''
+      Enables LSP features.
+    '';
+
+    lspFeatures.chunks = defaultNullOpts.mkStr "curly" ''
+      Sets chunk delimiter style.
+    '';
+
+    lspFeatures.languages =
+      defaultNullOpts.mkListOf types.str
+        [
+          "r"
+          "python"
+          "julia"
+          "bash"
+          "html"
+        ]
+        ''
+          List of supported languages.
+        '';
+
+    lspFeatures.diagnostics.enabled = defaultNullOpts.mkBool true ''
+      Enables diagnostics.
+    '';
+
+    lspFeatures.diagnostics.triggers = defaultNullOpts.mkListOf types.str [ "BufWritePost" ] ''
+      Sets triggers for diagnostics.
+    '';
+
+    lspFeatures.completion.enabled = defaultNullOpts.mkBool true ''
+      Enables completion.
+    '';
+
+    codeRunner.enabled = defaultNullOpts.mkBool false ''
+      Enables or disables the code runner.
+    '';
+
+    codeRunner.default_method =
+      defaultNullOpts.mkEnum
+        [
+          "molten"
+          "slime"
+        ]
+        null
+        ''
+          Sets the default code runner method. Either "molten", "slime", or `null`.
+        '';
+
+    codeRunner.ft_runners = defaultNullOpts.mkAttrsOf types.str { } ''
+      Specifies filetype to runner mappings.
+    '';
+
+    codeRunner.never_run = defaultNullOpts.mkListOf types.str [ "yaml" ] ''
+      List of filetypes that are never sent to a code runner.
+    '';
+  };
+
+  settingsExample = {
+    debug = false;
+    closePreviewOnExit = true;
+    lspFeatures = {
+      enabled = true;
+      chunks = "curly";
+      language = [
+        "r"
+        "python"
+        "julia"
+      ];
+      diagnostics = {
+        enabled = true;
+        triggers = [ "BufWritePost" ];
+      };
+      completion = {
+        enabled = true;
+      };
+    };
+    codeRunner = {
+      enabled = false;
+      default_method = "vim-slime";
+      ft_runners = {
+        python = "molten";
+      };
+      never_run = [ "yaml" ];
+    };
+  };
+
+  extraPlugins = with pkgs.vimPlugins; [
+    nvim-lspconfig
+    otter-nvim
+  ];
```

```suggestion

  extraPlugins = with pkgs.vimPlugins; [
    nvim-lspconfig
    otter-nvim
  ];
```

Such plugin dependencies should not be added in nixvim.
Either they are strictly required to run the plugin and, in this case, we would need to upstream this constraint in nixpkgs, or they are not. Then, we could add a warning/option for telling the user that this integrations are possible.

From my testing, it appears that only `nvim-lspconfig` is strictly required.

---

I will investigate nixpkgs then! When running locally, I would see quarto fail to build based on an otter.nvim dependency, but it should not be strictly required. Agreed.

---

https://github.com/NixOS/nixpkgs/pull/354634

---

Update: Still waiting on this to be in nixos-unstable.

---

I see this is now available in the nixos-unstable and nixpkgs-unstable branches, what else needs to be done to propagate those changes here? @GaetanLepage

---

I have just merged the `flake.lick` bump. It should work now.